src/qgis_geonode/conf.py

```diff
@@ -251,5 +253,146 @@ def is_current_connection(self, connection_id: uuid.UUID):
+                settings.setValue(
+                    "temporal_extent_start",
+                    current_filters.temporal_extent_start.toString(QtCore.Qt.ISODate),
+                )
+            else:
+                settings.setValue("temporal_extent_start", None)
+            if current_filters.temporal_extent_end is not None:
+                settings.setValue(
+                    "temporal_extent_end",
+                    current_filters.temporal_extent_end.toString(QtCore.Qt.ISODate),
+                )
+            else:
+                settings.setValue("temporal_extent_end", None)
+
+            if current_filters.publication_date_start is not None:
+                settings.setValue(
+                    "publication_date_start",
+                    current_filters.publication_date_start.toString(QtCore.Qt.ISODate),
+                )
+            else:
+                settings.setValue("publication_date_start", None)
+            if current_filters.publication_date_end is not None:
+                settings.setValue(
+                    "publication_date_end",
+                    current_filters.publication_date_end.toString(QtCore.Qt.ISODate),
+                )
+            else:
+                settings.setValue("publication_date_end", None)
+            if current_filters.spatial_extent is not None:
+                settings.setValue(
+                    "spatial_extent_north", current_filters.spatial_extent.yMaximum()
+                )
+                settings.setValue(
+                    "spatial_extent_south", current_filters.spatial_extent.yMinimum()
+                )
+                settings.setValue(
+                    "spatial_extent_east", current_filters.spatial_extent.xMaximum()
+                )
+                settings.setValue(
+                    "spatial_extent_west", current_filters.spatial_extent.xMinimum()
+                )
+            settings.setValue("sort_by_field", current_filters.ordering_field.value)
+            settings.setValue("reverse_sort_order", current_filters.reverse_ordering)
+
+    def get_current_search_filters(self) -> models.GeonodeApiSearchParameters:
+        with qgis_settings(
+            f"{self.BASE_GROUP_NAME}/current_search_filters"
+        ) as settings:
+            resources_types = []
+            temporal_extent_start = None
+            temporal_extent_end = None
+            publication_date_start = None
+            publication_date_end = None
+            spatial_extent = None
+
+            if settings.value("resource_types_vector", False, type=bool):
+                resources_types.append(models.GeonodeResourceType.VECTOR_LAYER)
+            if settings.value("resource_types_raster", False, type=bool):
+                resources_types.append(models.GeonodeResourceType.RASTER_LAYER)
+            if settings.value("resource_types_map", False, type=bool):
+                resources_types.append(models.GeonodeResourceType.MAP)
```

Comment:
These should default to `True` rather than `False`.

Otherwise, the very first time that a user uses the plugin they will all be False and the user will see a disabled search button. the user will then need to go to the advanced section and enable one of these explicitly. This seems confusing for first time users.

```suggestion
            if settings.value("resource_types_vector", True, type=bool):
                resources_types.append(models.GeonodeResourceType.VECTOR_LAYER)
            if settings.value("resource_types_raster", True, type=bool):
                resources_types.append(models.GeonodeResourceType.RASTER_LAYER)
            if settings.value("resource_types_map", True, type=bool):
                resources_types.append(models.GeonodeResourceType.MAP)
```